Add package no-emoji

ecraven opened the pull request:
### Brief summary of what the package does

Replace emoji by :emoji-name:

### Direct link to the package repository

https://github.com/ecraven/no-emoji

### Your association with the package

I am the author.

### Checklist

Please confirm with `x`:

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses). 
- [x] I've read [CONTRIBUTING.md](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.md)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.md](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.md)

purcell commented:
Thanks, looks good. Quick feedback:

- You should `;;;###autoload` the minor mode
- Don't define a face with only one colour: it is certain to look wrong on many/most users' screens. Instead, best practice is to define a face which `:inherit`s a built-in face, and then the user can customise it freely, but it will work with any theme if uncustomised.
- The code replaces the display table unconditionally, so will clash with other libraries which rely on display tables, e.g. my `page-break-lines` mode. It would be better to create a new display table only if there is not one already set.

ecraven commented:
Thanks for the thorough review!

I hope I have addressed all of these points:
- Autoload cookie added.
- The face now inherits from `dired-header'.
- I've taken the liberty to look at your code, and doing the same thing. I update an existing display table (though this will overwrite other modes' display-table settings if they conflict).
I've also refactored into a new function `no-emoji-displayable-unicode-name', so this can be re-defun'd if the actual format (:name:) is not what the user wishes to see.

Thanks for taking the time to review this ;)

purcell commented:
Great - thanks for your patience! Merged now.
